examples/AdvancedOperations/AddDynamicPageFeed.php

```diff
@@ -0,0 +1,487 @@
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $campaignId,
+        int $adGroupId
+    ) {
+        $dsaPageUrlLabel = "discounts";
+
+        // Creates the page feed details. This code example creates a new feed, but you can
+        // fetch and re-use an existing feed.
+        $feedDetails = self::createFeed($googleAdsClient, $customerId);
+        self::createFeedMapping($googleAdsClient, $customerId, $feedDetails);
+        self::createFeedItems($googleAdsClient, $customerId, $feedDetails, $dsaPageUrlLabel);
+
+        // Associates the page feed with the campaign.
+        self::updateCampaignDsaSetting($googleAdsClient, $customerId, $feedDetails, $campaignId);
+
+        // Optional: Targets web pages matching the feed's label in the ad group.
+        self::addDsaTarget($googleAdsClient, $customerId, $adGroupId, $dsaPageUrlLabel);
+
+        printf("Dynamic page feed setup is complete for campaign ID %d.%s", campaignId, PHP_EOL);
+    }
+
+    /**
+    * Creates a feed.
+    *
+    * @param GoogleAdsClient googleAdsClient the Google Ads API client
+    * @param int customerId the client customer ID in which to create the feed
+    * @return array the names and IDs of feed attributes
+    */
+    private static function createFeed(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId
+    ) {
+        // Creates a URL attribute.
+        $urlAttribute = new FeedAttribute([
+            'type' => FeedAttributeType::URL_LIST,
+            'name' => new StringValue(['value' => "Page URL"])
+        ]);
+
+        // Creates a label attribute.
+        $labelAttribute = new FeedAttribute([
+            'type' => FeedAttributeType::STRING_LIST,
+            'name' => new StringValue(['value' => "Label"])
+        ]);
+
+        // Creates the feed.
+        $feed = new Feed([
+            'name' => new StringValue(['value' => 'DSA Feed #' . uniqid()]),
+            'attributes' => [$urlAttribute, $labelAttribute],
+            'origin' => FeedOrigin::USER
+        ]);
+
+        // Creates a feed operation for creating a feed.
+        $feedOperation = new FeedOperation();
+        $feedOperation->setCreate($feed);
+
+        // Issues a mutate request to add the feed.
+        $feedServiceClient = $googleAdsClient->getFeedServiceClient();
+        $feedResponse = $feedServiceClient->mutateFeeds($customerId, [$feedOperation]);
+
+        $feedResourceName = $feedResponse->getResults()[0]->getResourceName();
+        printf("Added feed named '%s'.%s", $feedResourceName, PHP_EOL);
+
+        return self::getFeed($googleAdsClient, $customerId, $feedResourceName);
+    }
+
+    /**
+     * Retrieves details about a feed.
+     *
+     * @param GoogleAdsClient googleAdsClient the Google Ads API client
+     * @param int customerId the client customer ID
+     * @param string feedResourceName the resource name of the feed
+     * @return array the names and IDs of feed attributes
```

Comment:
This associative array doesn't contain only names and IDs?
And from what I see, it returns the feed attributes of only one item, right?
So there should be only one name and ID?

(Same for the PHPDoc of the calling function and other places below)

Reply:
Not sure about this: it contains a list of feed attributes, indexed by name, and the resource name.

Do you have a better suggestion for this?

Reply:
Yes, the part that looks confusing to me is that this array also contains the resource name of the "feed".
How about changing the PHPDoc to something like this:

@return array the feed details containing the feed's resource name and the feed attributes' names and values

It's more verbose but it reflects better the structure of the returned value itself, which should be less confusing and more maintainable in the future.

Reply:
Done, thanks!